webrender/src/device/gfx/device.rs

```diff
@@ -154,6 +233,12 @@ pub struct Device<B: hal::Backend> {
     fbos: FastHashMap<FBOId, Framebuffer<B>>,
     rbos: FastHashMap<RBOId, DepthBuffer<B>>,
     descriptor_pools: SmallVec<[DescriptorPools<B>; 1]>,
+    descriptors_per_draw: SmallVec<[FastHashMap<DescriptorSetResources, (usize, usize)>; 1]>,
```

Comment:
I think the naming here isn't the best as well, and I have to rename this. `descriptors_per_draw` is intended to hold the mapping between the resources and their corresponding descriptors. The `SmallVec`is here because usually we need to have this mapping for each frame because we can't map a descriptor set in multiple frames (this can lead to validation layer errors).